### PR TITLE
Update url in crypto flow with state

### PR DESCRIPTION
**Pull request by SudharakaP**
Resolve https://github.com/opencollective/opencollective/issues/6177

[crypto-url-sync.webm](https://user-images.githubusercontent.com/12435965/202890770-38af8231-fee3-46a8-b3d7-e69ac3816a6d.webm)

### CODE REVIEW

File: components/contribution-flow/StepDetailsCrypto.js [hunk 1]
```diff
@@ -123,12 +127,12 @@ const StepDetailsCrypto = ({ onChange, data, collective }) => {
         inputMode="decimal"
         defaultValue={amount}
         onChange={({ target }) => {
-          const amount = parseFloat(target.value);
+          const amount = parseFloat(target.value).toFixed(2);
```
Comment (Betree) [inline components/contribution-flow/StepDetailsCrypto.js:130]: I'm surprised by this, do we really want to limit to 2 decimals? Let's take BTC, 0.01BTC ~= $150 therefore wouldn't we prevent contributions under $150? Looking at the issue I don't really get what it's trying to solve

File: lib/UrlQueryHelper.js [hunk 2]
```diff
@@ -93,6 +93,18 @@ export default class UrlQueryHelper {
...
+    /**
+     * @param {string} str
+     * @returns {number}
+     */
+    float: str => {
+      if (isNil(str) || !str.length) {
+        return null;
+      } else {
+        const result = parseFloat(str);
+        return isNaN(result) || result;
```
Comment (Betree) [inline lib/UrlQueryHelper.js:105]: Let's not return a boolean, which would equal to `1.0`

```suggestion
        return isNaN(result) ? null : result;
```

Comment (SudharakaP) [reply lib/UrlQueryHelper.js]: Done. 👍🏽

File: components/contribution-flow/StepDetailsCrypto.js (hunk 1 above)
Comment (SudharakaP) [reply components/contribution-flow/StepDetailsCrypto.js]: Yes, this seems to be a mistake, thanks for pointing it out. I've corrected this.

**Review by Betree — APPROVED** [review]
Haven't tried but the code looks good, ok for me to merge if it works as intended

**Pull request merged**